### PR TITLE
Add account.update()

### PR DESCRIPTION
**Pull request by finnp**
Hey,

I've implemented `account.update()` for #52. The tests are passing. However I haven't tested it with an an actual hoodie server yet. So I'm not sure if the integration tests are actually correct.

But I thought I PR this to get some feedback.

Best,
Finn

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -48,6 +48,7 @@ account.on('signout', redirectToHome)
 - [account.on](#accounton)
 - [account.one](#accountone)
 - [account.off](#accountoff)
+- [account.update](#accountupdate)
```
Comment (gr2m) [inline README.md]: can you please move it below `account.fetch` / above `account.profile.get`?

File: README.md [hunk 2]
```diff
@@ -859,6 +860,73 @@ hoodie.off('connectionstatus:disconnected', showNotification)
...
+  </tr>
+</table>
+
+Example
+
+```js
+account.update({username: 'treetrunks'}).then(function (properties) {
+  alert('You are now known as ' + properties.username)
+})
+```
```
Comment (gr2m) [inline README.md]: can you please move it below account.fetch / above account.profile.get?

File: index.js [hunk 3]
```diff
@@ -35,6 +35,7 @@ function Account (options) {
     on: events.on.bind(this, state),
     one: events.one.bind(this, state),
     off: events.off.bind(this, state),
-    validate: require('./lib/validate').bind(this, state)
+    validate: require('./lib/validate').bind(this, state),
+    update: require('./lib/update').bind(this, state)
```
Comment (gr2m) [inline index.js]: sorry for the nitpicking, but could you please move `update` below `fetch` here as well?

File: test/unit/update-test.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  update({
+    cacheKey: 'cacheKey123',
+    url: 'http://example.com',
+    account: {
+      username: 'bakingpies',
+      session: {
+        id: 'abc1234'
+      }
+    }
+  }, {
```
Comment (gr2m) [inline test/unit/update-test.js]: can you please store the first argument in `var state = { ... }` and then do `update(state, { username: 'treetrunks' })`, that will make it more clear

**Comment by gr2m** [comment]
Amazing work, thanks  a bunch! Commit messages are perfect, too :+1: Could you work in my comments, then it’s perfect. If you are too busy then I can do it for you, but there is no rush :)

**Comment by finnp** [comment]
Okay, I think I've changed everything that you asked for :relaxed:

**Comment by gr2m** [comment]
:clap: merged via d8fca30

**Pull request closed without merging**